compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -135,29 +103,8 @@ static MatmulNarrowSizes getMatmulNarrowSizes(ShapedType outType,
         map.getResultPosition(getAffineDimExpr(dimPos, linalgOp->getContext()))
             .value());
   };
-  int64_t M, N;
-  switch (opType) {
-  case ContractionOpType::kMatmul:
-  case ContractionOpType::kBatchMatmul: {
-    M = getOutputSizeAtDimPos(cDims.m[0]);
-    N = getOutputSizeAtDimPos(cDims.n[0]);
-    break;
-  }
-  case ContractionOpType::kVecmat:
-  case ContractionOpType::kBatchVecmat: {
-    M = 1;
-    N = getOutputSizeAtDimPos(cDims.n[0]);
-    break;
-  }
-  case ContractionOpType::kMatvec:
-  case ContractionOpType::kBatchMatvec: {
-    M = getOutputSizeAtDimPos(cDims.m[0]);
-    N = 1;
-    break;
-  }
-  case ContractionOpType::kInvalid:
-    return MatmulNarrowSizes();
-  }
+  int64_t M = cDims.m.empty() ? 1 : getOutputSizeAtDimPos(cDims.m[0]);
+  int64_t N = cDims.n.empty() ? 1 : getOutputSizeAtDimPos(cDims.n[0]);
```

Comment:
I understand the logic is about vecmat/matvec, but it would be good to add a comment for other developers new to this code. Can you add a comment about why they are set to `1` when the lists are empty?